frontend/tet/admin/src/components/editor/EditById.tsx

```diff
@@ -1,25 +1,47 @@
-import * as React from 'react';
-
-import { useQuery } from 'react-query';
-import { BackendEndpoint } from 'tet/admin/backend-api/backend-api';
-import PageLoadingSpinner from 'shared/components/pages/PageLoadingSpinner';
-import { TetPostings } from 'tet/admin/types/tetposting';
+import React, { useContext, useState, useEffect } from 'react';
+import TetPosting from 'tet/admin/types/tetposting';
 import Editor from 'tet/admin/components/editor/Editor';
-import { TetEvent } from 'tet/admin/types/linkedevents';
-import { eventToTetPosting } from 'tet/admin/backend-api/transformations';
+import { $Heading, $HeadingContainer } from 'tet/admin/components/jobPostings/JobPostings.sc';
+import PreviewWrapper from 'tet/admin/components/editor/previewWrapper/PreviewWrapper';
+import PostingContainer from 'tet/shared/src/components/posting/PostingContainer';
+import { PreviewContext } from 'tet/admin/store/PreviewContext';
+import Container from 'shared/components/container/Container';
+import BackButton from 'tet/admin/components/BackButton';
 
 type EditByIdProps = {
-  id: string;
+  title: string;
+  data: TetPosting;
 };
 
-const EditById: React.FC<EditByIdProps> = ({ id }) => {
-  const { isLoading, data } = useQuery<TetEvent>(`${BackendEndpoint.TET_POSTINGS}${id}`);
+const EditById: React.FC<EditByIdProps> = ({ title, data }) => {
+  const { showPreview, tetPosting, getTemplateData } = useContext(PreviewContext);
+  const [isInitialRender, setIsInitialRender] = useState(true);
 
-  if (isLoading) {
-    return <PageLoadingSpinner />;
-  }
+  useEffect(() => {
+    //If initial, use data from query and not from previewContext
+    if (isInitialRender) setIsInitialRender(false);
+  }, []);
 
-  return data ? <Editor initialValue={eventToTetPosting(data)} /> : <>Not found.</>;
+  const templateData = getTemplateData();
+
+  if (showPreview) {
+    return (
+      <PreviewWrapper>
+        <PostingContainer posting={templateData} />
+      </PreviewWrapper>
+    );
+  }
+  return (
+    <>
+      <Container>
+        <BackButton />
+        <$HeadingContainer>
+          <$Heading>{title}</$Heading>
+        </$HeadingContainer>
+        <Editor initialValue={isInitialRender ? data : tetPosting} />
```

Comment:
I think you could use `initialData` or `placeholderData` of [react-query](https://react-query.tanstack.com/guides/initial-query-data)